PGM/src/main/java/tc/oc/pgm/commands/MapCommands.java

```diff
@@ -41,24 +51,97 @@
 import tc.oc.pgm.rotation.RotationState;
 import tc.oc.pgm.teams.TeamFactory;
 
-public class MapCommands {
+public class MapCommands implements Commands {
+    private final UserFinder userFinder;
+    private final SyncExecutor syncExecutor;
+    private final IdentityProvider identityProvider;
+    private final ComponentRenderContext renderer;
+
+    @Inject MapCommands(UserFinder userFinder, IdentityProvider identityProvider, SyncExecutor syncExecutor, ComponentRenderContext renderer) {
+        this.userFinder = userFinder;
+        this.identityProvider = identityProvider;
+        this.syncExecutor = syncExecutor;
+        this.renderer = renderer;
+    }
+
     @Command(
         aliases = {"maplist", "maps", "ml"},
         desc = "Shows the maps that are currently loaded",
-        usage = "[page]",
+        usage = "[-a author] [-g gamemode] [page]",
         min = 0,
-        max = 1,
+        max = 2,
+        flags = "a:g:",
         help = "Shows all the maps that are currently loaded including ones that are not in the rotation."
     )
     @CommandPermissions("pgm.maplist")
-    public static void maplist(CommandContext args, final CommandSender sender) throws CommandException {
-        final Set<PGMMap> maps = ImmutableSortedSet.copyOf(new PGMMap.DisplayOrder(), PGM.getMatchManager().getMaps());
+    public void maplist(final CommandContext args, final CommandSender sender) throws CommandException {
+        final Identity senderIdentity = identityProvider.createIdentity(sender);
+        final MapDoc.Gamemode gamemode = parseGamemode(args.getFlag('g'));
+        final String author = args.getFlag('a');
+        if(args.getFlag('a') != null) {
+            syncExecutor.callback(
+                userFinder.findUser(sender, author, UserFinder.Scope.ALL, UserFinder.Default.NULL),
+                CommandFutureCallback.onSuccess(sender, result -> {
+                        BaseComponent displayMsg = args.hasFlag('g') ? new TranslatableComponent("command.map.mapList.displayByBoth",
```

Comment:
use gamemode != null instead of  args.hasFlag('g')
i'd split the lines so that each "new TranslatableComponent" starts a new line